File: microservices/investment/src/main/java/com/illunex/invest/investment/persistence/entity/EvaluateJudge.java
package com.illunex.invest.investment.persistence.entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "evaluate_judge")
public class EvaluateJudge {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idx;
    Long userIdx;
    String name;
    String company;
    String rank;
    String comment;
    String imgUrl;
    String status;
    Integer finalScore;
    LocalDateTime evaluateDate;

    @ManyToOne
    @JoinColumn(name = "evaluate_idx")
    Evaluate evaluate;

    @OneToMany(mappedBy = "judge", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    List<EvaluateJudgeScore> scoreList;
}

File: microservices/invest/src/main/kotlin/com/illunex/invest/invest/persistence/entity/EvaluateJudge.kt
package com.illunex.invest.invest.persistence.entity

import javax.persistence.*

@Entity
@Table(name = "evaluate_judge")
data class EvaluateJudge(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idx: Long? = null,
    var userIdx: Long? = null,
    var name: String? = null,
    var comment: String? = null,
    var point: Int? = 0,
    var rank: String? = null,
    var imgUrl: String? = null,
    var status: String? = null,

    @ManyToOne @JoinColumn(name = "evaluate_idx")
    var evaluate: Evaluate?
) {
    constructor() : this(null, null, null, null, null, null, null, null, null )
}

